numpy/lib/_index_tricks_impl.py

```diff
@@ -682,16 +684,24 @@ class ndindex:
     (1, 1, 0)
     (2, 0, 0)
     (2, 1, 0)
+    
+    Notes
+    -----
+    New in version X.Y.Z.
+    As of NumPy 2.3.0.dev0, this iterator is implemented using `itertools.product`
+    from Python's standard library. This change provides significant improvements
+    in both performance and and memory efficiency, particularly for large iteration
```

Comment:
repeated `and`; we'll probably want to change the version numbering you use above since i.e., `2.3.0` has already been released